src/services/API/RootClient.js

```diff
@@ -34,6 +35,18 @@ client.interceptors.request.use(
   },
   (error) => {
     // handle the error
+    ErrorHandling.handleAPIErrors(error);
+    return Promise.reject(error);
+  }
+);
+
+// handle errors in responses for API requests
+client.interceptors.response.use(
+  (config) => {
+    return config;
+  },
+  (error) => {
+    ErrorHandling.handleAPIErrors(error);
```

Comment:
Are you sure both the types of errors should be handled by the same thing? The request hasn't happened only in the upper one - so, there won't be any API errors no? the error there is just any error in the interceptor code right?